transformer/mappings.py

```diff
@@ -112,20 +84,35 @@ def title(self, value):
     def type(self, value):
         return value.split('note_',1)[1]
 
-    @odin.map_list_field(from_field='subnotes', to_field='subnotes', to_list=True)
-    def subnotes(self, value):
-        if value:
-            return (ArchivesSpaceSubnoteToSubnote.apply(v) for v in value)
-
-    @odin.map_field(from_field='content', to_field='subnotes', to_list=True)
-    def subnotes(self, value):
-        if value:
-            return Subnote(type='text', content=value)
+    def map_subnotes(self, value):
+        if value.jsonmodel_type in ['note_orderedlist', 'note_definedlist']:
+            return Subnote(type=value.jsonmodel_type.split('note_')[1], content=value.items)
+        elif value == 'note_bibliography':
+            data = []
+            data.append(Subnote(type='text', content=value.content))
+            data.append(Subnote(type='orderedlist', content=value.items))
+            return data
+        elif value.jsonmodel_type == 'note_index':
+            data = []
+            l = [{'label': i.get('type'), 'value': i.get('value')} for i in value.items]
+            data.append(Subnote(type='text', content=value.content))
+            data.append(Subnote(type='definedlist', content=l))
+            return data
+        elif value.jsonmodel_type == 'note_chronology':
+            m = [{'label': i.get('event_date'), 'value': ', '.join(i.get('events'))} for i in value.items]
+            return Subnote(type='definedlist', content=m)
+        else:
+            return Subnote(type='text', content=value.content
+                           if isinstance(value.content, list) else [value.content])
```

Comment:
Here I've moved the `ArchivesSpaceSubnoteToSubnote` mapping logic down into the `ArchivesSpaceNoteToNote` class. This allows me to use the whole scope of the note, which is useful given the way that content is distributed in AS notes.